sycl/plugins/cuda/pi_cuda.cpp

```diff
@@ -3038,7 +3038,10 @@ pi_result cuda_piextKernelSetArgMemObj(pi_kernel kernel, pi_uint32 arg_index,
                                        const pi_mem *arg_value) {
 
   assert(kernel != nullptr);
-  assert(arg_value != nullptr);
```

Comment:
Could we keep this check? You are dereferencing it below, so it still seems valid. Same goes for the HIP changes.

Reply:
thansk @steffenlarsen , done